localstack/services/apigateway/provider.py

```diff
@@ -515,6 +515,100 @@ def _delete_children(resource_to_delete: str):
         parent_id = moto_resource.parent_id
         api_resources[parent_id].remove(resource_id)
 
+    def update_integration(
+        self,
+        context: RequestContext,
+        rest_api_id: String,
+        resource_id: String,
+        http_method: String,
+        patch_operations: ListOfPatchOperation = None,
+    ) -> Integration:
+        # XXX: THIS IS NOT A COMPLETE IMPLEMENTATION, just the minimum required to get tests going
+        # TODO: validate patch operations
+
+        moto_rest_api = get_moto_rest_api(context, rest_api_id)
+        moto_resource = moto_rest_api.resources.get(resource_id)
+        if not moto_resource:
+            raise NotFoundException("Invalid Resource identifier specified")
+
+        moto_method = moto_resource.resource_methods.get(http_method)
+        if not moto_method:
+            raise NotFoundException("Invalid Method identifier specified")
+
+        integration = moto_method.method_integration
+
+        for patch_operation in patch_operations:
+            op = patch_operation.get("op")
+            path = patch_operation.get("path")
+            if "/requestParameters" not in path:
+                raise BadRequestException(
+                    f"Invalid patch path  '{path}' specified for op '{op}'. Must be one of: [/requestParameters]"
+                )
+            if op != "add":
+                raise BadRequestException(
+                    f"Invalid patch path  '{path}' specified for op '{op}'. Please choose supported operations"
+                )
```

Comment:
I think this is too strict, and would raise Exceptions wen AWS doesn't.

Reply:
too strict because because we don't implement other operations. I followed the same idea of `update_resource` you did given your experience in the patch zone.

Reply:
The implementation is good, it's just that the exceptions raised in `update_resource` are raised by AWS too, and I'm worried it's not the case here.